### DISCUSSION TITLE
accessing android apis

### DISCUSSION THREAD

**Discussion by plugeit** [flet-dev/flet#1877] [General]
Feodor, I have some questions about android and ios apps that were not clear to me:
1 - All application logic can be done in python with flet, in this case can I use PYPI libraries?
2 - Is there a script for accessing native resources such as camera, notifications, gps, sensors? mainly android native resources
3 - Is my python code safe in the APK?

I ask this because I wouldn't want to use an API for my application, like web development with Flet.

**Comment by SKbarbon** [comment]
@icanhackyou may you allow me to answer your questions.

1. Yes, you can use PYPI packages in your flet mobile app. All you need to do is to add the package name in the requirements.txt file.

2. Yes, you can access some of mobile resources such as camera and microphone. Flet doesn’t support that directly from the flet flutter API, but flet runs it self using the kivy python run-time, which is supporting a python library named [pyler](https://github.com/kivy/plyer).

3. Nothing is 100% safe, but flet compile it self to .pyc files (a byte code). Which is very hard to reverse engineering it.

**Reply by FeodorFitsner** [reply]
Thanks @SKbarbon! @icanhackyou please avoid [duplicating](https://github.com/flet-dev/flet/discussions/1750) the discussions. If the intention was to draw attention to the topic then you succeeded, but it's not cool to have the inbox full of GitHub notifications with the same questions.

**Reply by plugeit** [reply]
Yes, the intention was to draw the attention and, I want to apologize if you got annoyed by it. I did it because no one responded to my earlier question related to accessing the SMS API on android 
i also want to ask there is a python module named py4a [https://pythonspot.com/tag/py4a/] which allows accessing features like camera, SMS, etc. Is there any way to use that.
